### ISSUE TITLE
Rename `Taichi THREE` to `THREE.py`?

### ISSUE THREAD

**Issue by archibate** [taichi-dev/taichi_three#21]
As Taichi THREE grows, it become more and more mature.
I'm wondering if `taichi_three` is a convinent and eye-catching name.
I think we'd be ambitous, to push this project not only an extension library of Taichi, but a standard soft rendering pipeline based on Taichi.
It currently supports shadow mapping, normal maps, PBR materials, OBJ loader, etc. which is totally a small but strong bird (麻雀虽小, 五脏俱全). We even have set up a documentaion site at: http://t3.142857.red/ (WIP).
Compared to other real rendering pipelines like https://github.com/stemkoski/three.py, our advantange is flexibility. Not to say we're going to support path tracing scheme.
So, I'd like to rename Taichi THREE to THREE.py, to claim that we aren't just an extension library, but a complete rendering pipeline.
We'll seeing it finally grows up to become a industrial renderer, like Eevee and Cycles. We will finally see an option named `THREE.py`, right below `Cycles (工作台)` in Blender's "bake" menu!
`three.Camera` also sounds cooler than `t3.Camera` to me.
WDYT about this cool ambition? Inputs are welcome! @victoriacity @Zony-Zhao @yuanming-hu

**Comment by Iteravse** [comment]
new document looks great~
may i more prefer rename to Taichi3,there meaning in chinese's taichi born everything 
if just rename to THREE.py,it sounds just a python 3d library,but i hope some day may we can independent with python,use another more efficiency method programme,let it become a new platform
i hope this platform more powerful,effectively,conveniency than any exists engine enough to realize any vision in the world
but i think if we really want to develop and gorw stronger taichi3d as fantastic Tools even better any other Engine in this world like Unreal,Frostbite,Unity3d,we have one advantage over them,a stronger core arithmetic 'taichi'
BTW i invent a board game to find a new rule let human challenge deepmind artificial intelligent,and i try to realize that game graphical interface by taichi,so i more prefer Taichi3 become a unique engine not a just python library,just my opinion,the hope can be helpful for you

**Comment by victoriacity** [comment]
Here're some of my ideas:

Naming - I feel that Taichi has been making the distinction of being its own programming language. For example, we almost always say "using the Taichi programming language" instead of "using the Taichi package in Python". When talking about a package of Python, one may naturally think of the NumPy/SciPy stack, which is the case for PyTorch, TensorFlow, as well as three.py. In this regard, `taichi_three` would still be a good name since it is "the" 3D rendering extension of the Taichi language. However this will also relate to the long-term plan for Taichi in whether we tend to go closer to Python or depart farther from Python.

I more envision taichi_three to be a more research-oriented renderer than a production-oriented renderer. The greatest benefit of taichi_three is flexibility, which directly accelerates graphics research, for example, imagine reproducing a SIGGRAPH rendering paper in less than 100 lines. A Blender extension will be great -- one of the most common research renderer, Mitsuba, also has a Blender extension.

**Comment by archibate** [comment]
Thank for all the discussions here! Here's the recent progress on this topic:

The core is completely re-written after Taichi THREE v0.0.9 is released making the API more intuitive and much easier to maintain in hope for make it available to everyone.
Don't worry, many of the good things from legacy Taichi THREE are left, like cook-torrance, mesh editing nodes, OBJ importer, marching cube... but some of the very core parts, including the triangle rasterizer, is completely thrown away and replaced by a more efficient algorithm that is friendly to GPUs when faces are a bit large. The new rasterizer also make compilation a bit faster and no more growing compilation time when there are a lot of models (reliefs #26). Also note that the camera system is completely re-rewritten (sorry @Zony-Zhao and @victoriacity!) and now we no longer have dead-lock (万向节死锁) at +Y and -Y.
The re-written renderer is renamed to `Tina` to celebrate the huge changes in its API and important refactors and steps as an answer to this issue, source of the name might be [`Seventina`](https://www.bilibili.com/video/BV1zt411U7ZE), one of my favorite song of harumakigohan :laughing:
Also thanks to the cleaned code structure, the renderer now also have a Blender integration work in progress, see [`Taichi-Blend`](https://github.com/taichi-dev/taichi_blend), and [video demo here](https://www.bilibili.com/video/BV17i4y157xx).
